homeassistant/components/whirlpool/climate.py

```diff
@@ -0,0 +1,189 @@
+"""Platform for climate integration."""
+import asyncio
+import logging
+
+import aiohttp
+from whirlpool.aircon import Aircon, FanSpeed as AirconFanSpeed, Mode as AirconMode
+from whirlpool.auth import Auth
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from .const import AUTH_INSTANCE_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+AIRCON_MODE_MAP = {
+    AirconMode.Cool: HVAC_MODE_COOL,
+    AirconMode.Heat: HVAC_MODE_HEAT,
+    AirconMode.Fan: HVAC_MODE_FAN_ONLY,
+}
+
+HVAC_MODE_TO_AIRCON_MODE = {v: k for k, v in AIRCON_MODE_MAP.items()}
+
+AIRCON_FANSPEED_MAP = {
+    AirconFanSpeed.Off: FAN_OFF,
+    AirconFanSpeed.Auto: FAN_AUTO,
+    AirconFanSpeed.Low: FAN_LOW,
+    AirconFanSpeed.Medium: FAN_MEDIUM,
+    AirconFanSpeed.High: FAN_HIGH,
+}
+
+FAN_MODE_TO_AIRCON_FANSPEED = {v: k for k, v in AIRCON_FANSPEED_MAP.items()}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    auth: Auth = hass.data[DOMAIN][config_entry.entry_id][AUTH_INSTANCE_KEY]
+    said_list = auth.get_said_list()
+    if not said_list:
+        _LOGGER.debug("No appliances found")
+        return
+
+    aircon = AirConEntity(said_list[0], auth)
+    async_add_entities([aircon], True)
+
+
+class AirConEntity(ClimateEntity):
+    """Representation of an air conditioner."""
+
+    _attr_fan_modes = [FAN_AUTO, FAN_HIGH, FAN_MEDIUM, FAN_LOW, FAN_OFF]
+    _attr_hvac_modes = [
+        HVAC_MODE_COOL,
+        HVAC_MODE_HEAT,
+        HVAC_MODE_FAN_ONLY,
+        HVAC_MODE_OFF,
+    ]
+    _attr_max_temp = 30
+    _attr_min_temp = 16
+    _attr_supported_features = (
+        SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE | SUPPORT_SWING_MODE
+    )
+    _attr_swing_modes = [SWING_HORIZONTAL, SWING_OFF]
+    _attr_target_temperature_step = 1
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, said, auth: Auth):
+        """Initialize the entity."""
+        self._aircon = Aircon(auth, said, self.schedule_update_ha_state)
```

Comment:
It looks like the callback is run in the event loop : https://github.com/abmantis/whirlpool-sixth-sense/blob/ca336173d3b5d9a13e7e4b0fa7ca998a9b71d729/whirlpool/eventsocket.py#L82

If so we should write the state directly instead

```suggestion
        self._aircon = Aircon(auth, said, self.async_write_ha_state)
```